movieApp.js

```diff
@@ -1,204 +1,279 @@
-  movieContainer.className = "movie-container";
-
-  //
-  //  Create an img element and set its src attribute
-  const img = document.createElement("img");
-  img.width = 200;
-  img.src = movie.poster_url;
-
-  // Create a button element
-  const readMoreBtn = document.createElement("button");
-  readMoreBtn.textContent = "Read More";
-  readMoreBtn.className = "accordion";
-
-  // starRating
-  const starRating = document.createElement("div");
-  starRating.className = "star-rating";
-  const numberOfStars = 5;
-  let stars = [];
-
-  for (let x = 0; x < numberOfStars; x++) {
-    const star = document.createElement("i");
-    star.className = "rating__star far fa-star";
-    starRating.appendChild(star);
-    stars.push(star);
-  }
-  const starClassActive = "rating__star fas fa-star";
-  const starClassInactive = "rating__star far fa-star";
-  stars.forEach((star, clickedStarIndex) => {
-    star.onclick = () => {
-      stars.forEach((star, starIndex) => {
-        if (starIndex <= clickedStarIndex) {
-          star.className = starClassActive;
-        } else {
-          star.className = starClassInactive;
-        }
+document.addEventListener("DOMContentLoaded", getMoviesData);
+
+async function getMoviesData() {
+  const response = await fetch(
+    "https://raw.githubusercontent.com/myselfankitaa/myselfankitaa.github.io/main/movies.json"
+  );
+  const moviesData = await response
+    .json()
+    .then(processMoviesData)
+    .catch((error) => (`ERROR !!!`, error))
+    .finally(() => console.log(`Done`));
+}
+
+function processMoviesData(movies) {
+  const nowStreaming = document.querySelector(".now-streaming");
+
+  function createMovieCard(movie) {
+    //Create a container to store movie details
+    const movieDataContainer = document.createElement("div");
+    movieDataContainer.className = "movieDataContainer";
+    const movieContainer = document.createElement("div");
+    movieContainer.className = "movie-container";
+
+    //  Create an img element and set its src attribute
+    const img = document.createElement("img");
+    img.width = 200;
+    img.src = movie.poster_url;
+
+    // Create a button element
+    const readMoreBtn = document.createElement("button");
+    readMoreBtn.textContent = "Read More";
+    readMoreBtn.className = "readMore";
+
+    // starRating
+    const starRating = document.createElement("div");
+    starRating.className = "star-rating";
+    const numberOfStars = 5;
+    const stars = [];
+
+    for (let x = 0; x < numberOfStars; x++) {
+      const star = document.createElement("i");
+      star.className = "rating__star far fa-star";
+      starRating.appendChild(star);
+      stars.push(star);
+    }
+    const starClassActive = "rating__star fas fa-star";
+    const starClassInactive = "rating__star far fa-star";
+    stars.forEach((star, clickedStarIndex) => {
+      star.addEventListener("click", () => {
+        stars.forEach((star, starIndex) =>
+          starIndex <= clickedStarIndex
```

Comment:
Although this works perfectly fine, it is not customary in the JavaScript world to write a check like this, where the second part is two statements. 

In other words, the more common way of writing this would be: 

`star.className = (condition) ? ... : ...`

Reply:
I did it as suggested